File: Lab2/app/src/main/java/com/koroliuk/lab2/fragment/AlertFragment.java
package com.koroliuk.lab2.fragment;

import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import com.koroliuk.lab2.R;

public class AlertFragment extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        return new AlertDialog.Builder(requireContext())
                .setMessage(R.string.dialog_message)
                .setTitle(R.string.dialog_title)
                .setPositiveButton(getString(R.string.ok), (dialog, which) -> {} )
                .create();
    }

    public static String TAG = "AlertDialog";
}

File: Lab3/app/src/main/java/com/koroliuk/lab3/fragment/AlertFragment.kt
package com.koroliuk.lab3.fragment

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.koroliuk.lab3.R

class AlertFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog =
        AlertDialog.Builder(requireContext())
            .setMessage(getString(R.string.dialog_message))
            .setPositiveButton(getString(R.string.button_ok)) { _, _ -> }
            .create()

    companion object {
        const val TAG = "AlertDialog"
    }
}